uwblueprint/project-read-frontend
# [4/5] Set up registration form inputs & submission

### Notion ticket link



[Set up reg form inputs & submit to POST /families](https://www.notion.so/uwblueprintexecs/9cb454262ed84c6299c0682c23f92ec5?v=1076bd30711a4a01838becee05f8b5a3&p=3772e0140c85481e93c6394fff197608)

lots of next steps for this, will be adding follow-up tasks on notion 🏃‍♀️ 



### Implementation description

- filled in the `RegistrationForm` component with the default & dynamic fields for families, parents, children, and guests
  - it uses a `FormFieldGroup` component for each of the different "parts" of data ("default" and "dynamic" fields for each type of student) that go into the POST /families endpoint for ease of structuring the payload
  - so far it defaults to one child and one guest – adding/removing children & guests will come in a follow-up task
- created a `FormFieldGroup` component that renders inputs based on the provided fields
- added the POST family endpoint to the api utils



### Steps to test

there will be an automated test in the next PR! to test manually:

1. `yarn start`
2. open the registration form
3. fill in some data & submit
  - if the info is invalid, an alert will show (form validations to come later)
  - otherwise it takes you back to the "Add a client" view, and your new family should show in the reg table!

<img width="1363" alt="Screen Shot 2021-04-29 at 9 31 12 PM" src="https://user-images.githubusercontent.com/37346399/116637560-3a822a00-a932-11eb-832c-d3de676a3397.png">



### What should reviewers focus on?

- code cleanliness
- is there a better way to write anything? let me know 😄 

### Checklist
